### PR TITLE
#3519 KYC changes for US investors

### PR DESCRIPTION
**Pull request by drakkein**
This is basic version of changes with current file uploader.

Some consulting with @Louisswryan and @andywalton are required before pushing forward.

### CODE REVIEW

File: packages/web/app/components/shared/UploadedFiles.tsx [hunk 1]
```diff
@@ -13,18 +12,14 @@ interface IProps {
...
-      <FormattedMessage id="shared-component.uploaded-files.title" />
-    </h6>
     {files.map(({ fileName }, index) => (
       <div data-test-id={`multi-file-upload-file-${fileName}`} key={index} className={styles.file}>
+        <img src={confirmIcon} className={styles.icon} />
         <span className={styles.text} title={fileName}>
           {fileName}
         </span>
-        <img src={confirmIcon} className={styles.icon} />
+        {/* TODO Add ability to download and remove file */}
```
Comment (drakkein) [inline packages/web/app/components/shared/UploadedFiles.tsx]: Right now backend is returning only file name. We need download and delete methods

**Comment by andywalton** [comment]
@drakkein - I Slacked you 3 small changes.

1. A bug that the uploader widget doesn't hide should you change your address or nationality away from US.
2. Layout change - to move the `Are you an accredited US investor?` to the end of the form above the Continue button
3. To add two h5 labels to the page with heading copy to define the `Personal information` from the `Address details` and the `Accreditation status` sections.

File: packages/web/app/lib/api/kyc/KycApi.interfaces.ts [hunk 2]
```diff
@@ -38,11 +42,29 @@ export const KycPersonSchema = Yup.object().shape({
...
   isPoliticallyExposed: Yup.bool(),
+  usState: Yup.string().when("country", {
+    is: val => val === ECountries.UNITED_STATES,
+    then: Yup.string().required(),
+    otherwise: Yup.string(),
+  }),
+});
+
+export const KycAdditionalDataSchema = Yup.object().shape({
+  isAccreditedUsCitizen: Yup.bool().when(["country", "nationality"], {
```
Comment (desfero) [inline packages/web/app/lib/api/kyc/KycApi.interfaces.ts]: Check the comment below about refactoring `when` a little bit

File: packages/web/app/components/shared/UploadedFiles.tsx [hunk 3]
```diff
@@ -13,18 +12,14 @@ interface IProps {
 
 export const UploadedFiles: React.FunctionComponent<IProps> = ({ files }) => (
   <div className={cn("c-uploaded-files", styles.uploadedFiles)}>
-    <h6>
-      {" "}
-      <FormattedMessage id="shared-component.uploaded-files.title" />
-    </h6>
     {files.map(({ fileName }, index) => (
       <div data-test-id={`multi-file-upload-file-${fileName}`} key={index} className={styles.file}>
+        <img src={confirmIcon} className={styles.icon} />
```
Comment (desfero) [inline packages/web/app/components/shared/UploadedFiles.tsx]: please add `alt=""`

File: packages/web/app/components/kyc/personal/Start.unsafe.tsx [hunk 4]
```diff
@@ -5,11 +5,14 @@ import { Col, Row } from "reactstrap";
 import { compose } from "redux";
 
 import {
+  EKycRequestType,
   IKycIndividualData,
-  KycIndividualDataSchemaRequired,
+  KycIndividualDataSchemaRequiredWithAdditionalData,
 } from "../../../lib/api/kyc/KycApi.interfaces";
 import { actions } from "../../../modules/actions";
+import { ENotificationText, ENotificationType } from "../../../modules/notifications/reducer";
```
Comment (desfero) [inline packages/web/app/components/kyc/personal/Start.unsafe.tsx]: maybe we can move `ENotificationText` and `ENotificationType` to `modules/notifications/types` to just not import from `reducer` inside component

File: packages/web/app/components/kyc/personal/Start.unsafe.tsx [hunk 5]
```diff
@@ -65,17 +71,21 @@ const HIGH_INCOME_VALUES = {
...
+  submitForm: (values: IKycIndividualData, skipContinue?: boolean) => void;
 }
 
 type IProps = IStateProps & IDispatchProps & FormikProps<IKycIndividualData>;
 
-const KYCForm = injectIntlHelpers<IProps & IKycIndividualData>(
+const KYCForm = injectIntlHelpers<IProps & IKycIndividualData & IDispatchProps>(
   ({ intl: { formatIntlMessage }, ...props }) => (
     <FormDeprecated>
+      <h5 className="mb-3">
```
Comment (desfero) [inline packages/web/app/components/kyc/personal/Start.unsafe.tsx]: just a nitpick for the future, from the semantic point of view we should just `fieldset` and `legend` here.

File: packages/web/app/components/shared/MultiFileUpload.module.scss [hunk 6]
```diff
@@ -121,3 +121,27 @@ $_dropzone-size-sm: 120px;
...
+}
+
+.uploader-info {
+  flex-shrink: 3;
+  margin-left: px-to-rem(12px);
+}
+
+.file-types {
+  font-family: $font-family-roboto;
+  font-size: px-to-rem(12px);
```
Comment (desfero) [inline packages/web/app/components/shared/MultiFileUpload.module.scss]: use `typography` mixin

File: packages/web/app/components/shared/MultiFileUpload.module.scss [hunk 7]
```diff
@@ -121,3 +121,27 @@ $_dropzone-size-sm: 120px;
...
+  line-height: 1.33;
+  color: $neu-shuttle-gray;
+}
+
+.upload-title {
+  font-family: $font-family-roboto;
+  font-size: px-to-rem(14px);
```
Comment (desfero) [inline packages/web/app/components/shared/MultiFileUpload.module.scss]: use typography mixin (probably the new one is needed)

File: packages/web/app/components/shared/Dropzone.module.scss [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  right: 0;
+  bottom: 0;
+  margin: -2px;
+  background-color: $loading-background;
+  color: $neu-white;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  z-index: 10;
```
Comment (desfero) [inline packages/web/app/components/shared/Dropzone.module.scss]: No need for `z-index` if you move `DropzoneSpinner` after `DropzoneContent`

File: packages/web/app/components/shared/Dropzone.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { storiesOf } from "@storybook/react";
+import React from "react";
+
+import { Dropzone } from "./Dropzone";
+
+storiesOf("Dropzone", module)
```
Comment (desfero) [inline packages/web/app/components/shared/Dropzone.stories.tsx]: 🎖

File: packages/web/app/components/shared/notification-widget/Notification.tsx [hunk 10]
```diff
@@ -40,14 +41,19 @@ const notificationTexts = {
...
-    <FormattedHTMLMessage
-      tagName="span"
+    <FormattedMessage
       id="notifications.not-accredited-investor"
       values={{
-        accreditationHref: externalRoutes.accreditationHelp,
-        accreditationLabel: getHostname(externalRoutes.accreditationHelp),
-        accreditationHrefSecond: externalRoutes.accreditationHelpSecond,
-        accreditationLabelSecond: getHostname(externalRoutes.accreditationHelpSecond),
+        accreditationHref: (
```
Comment (desfero) [inline packages/web/app/components/shared/notification-widget/Notification.tsx]: 🎖

File: packages/web/app/lib/api/kyc/KycApi.ts [hunk 11]
```diff
@@ -52,8 +52,7 @@ export class KycApi {
     return await this.httpClient.get<IKycBusinessData>({
       baseUrl: BASE_PATH,
       url: INDIVIDUAL_DATA_PATH,
-      // TODO: Move back when KYC flow for us investor is done
-      // responseSchema: KycIndividualDataSchema,
+      responseSchema: KycIndividualDataSchema,
```
Comment (desfero) [inline packages/web/app/lib/api/kyc/KycApi.ts]: thanks a lot for moving that back

**Comment by Moejoe90** [comment]
Will merge to test on `io`

**Pull request merged**